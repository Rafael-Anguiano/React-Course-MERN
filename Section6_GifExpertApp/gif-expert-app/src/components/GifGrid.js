import React /* { useState, useEffect } */ from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types'

const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { loading && <p>Loading...</p>}

            <div className='card-grid animate_animated animate__fadeInDown animate__delay-5s'>
                { 
                    images.map( (img) =>  (
                        <GifGridItem 
                        key={img.id} 
                        {...img}
                        />
                        ))
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifGrid
