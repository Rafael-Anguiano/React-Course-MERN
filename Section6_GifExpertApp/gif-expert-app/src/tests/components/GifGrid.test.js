import React from 'react'
import GifGrid from '../../components/GifGrid';
import { shallow } from "enzyme"
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom'
jest.mock('../../hooks/useFetchGifs');

describe('Test in GifGrid Component', () => {
    const category ="HxH"
    // let wrapper;

    // beforeEach(()=>{
    //     jest.clearAllMocks();
    //     wrapper = shallow(<GifGrid category={category} />)
    // })

    test('should match with snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={category} /> );
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should show items when images are charge useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://testing/anything.jpg',
            title: 'Gif Title'
        }, {
            id: 'ABC',
            url: 'https://testing/anything.jpg',
            title: 'Gif Title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length)
    })
    
})
