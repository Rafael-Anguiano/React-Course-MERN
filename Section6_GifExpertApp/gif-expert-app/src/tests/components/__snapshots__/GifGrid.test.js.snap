// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGrid Component should match with snapshot 1`] = `
<Fragment>
  <h3>
    HxH
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid animate_animated animate__fadeInDown animate__delay-5s"
  />
</Fragment>
`;
